FITist_project/user_profiles/views.py

```diff
@@ -36,11 +36,26 @@ def get_profile(request, username=None):
     if not username:
         return redirect('profiles:user_profile', username=curr_user.username)
     user = get_object_or_404(User, username=username)
-    if request.user == user:
+    if curr_user == user:
         return render(
             request,
             'user_profiles/profile_detail.html',
             {'user': user}
         )
     else:
         return redirect('profiles:user_profile', username=curr_user.username)
+
+
+def view_program(request, username=None):
+    curr_user = request.user
+    if not username:
```

Comment:
The username isn't required to view the profile (it just requires that someone is logged in) but it is required to view a workout for the current user. I did this to prevent someone from being able to see someone elses workouts.